### PR TITLE
Generate fake user data (bug 1119543)

### PR DESCRIPTION
**Pull request by washort**

### CODE REVIEW

File: mkt/api/models.py [hunk 1]
```diff
@@ -60,7 +71,3 @@ class Meta:
         db_table = 'oauth_nonce'
         unique_together = ('nonce', 'timestamp', 'client_key',
                            'request_token', 'access_token')
-
-
-def generate():
-    return os.urandom(64).encode('hex')
```
Comment (robhudson) [inline mkt/api/models.py]: Looks like this broke something and tests failed.

Comment (washort) [inline mkt/api/models.py]: Tests now pass.

**Comment by robhudson** [comment]
Looks like it needs a rebase as well. I can take a close look before finishing up today.

**Comment by washort** [comment]
rebased.

File: mkt/users/tasks.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import datetime
+from django.db import transaction
+
+import mkt
+from mkt.access.models import Group, GroupUser
+from mkt.api.models import Access
+from mkt.users.models import UserProfile
+
+
```
Comment (robhudson) [inline mkt/users/tasks.py]: This is in the `tasks.py` file but it's not using celery tasks?

File: mkt/users/tasks.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        else:
+            if oauth_key or oauth_secret:
+                raise ValueError("Specify both of oauth_key and oauth_secret, "
+                                 "or neither")
+            Access.create_for_user(profile)
+
+
+def generate_users(n):
+    for i in range(n):
+        create_user('testuser%s@example.com' % (i,), "Test User %d" % (i,))
```
Comment (robhudson) [inline mkt/users/tasks.py]: It doesn't look like this is used anywhere?

File: mkt/users/tasks.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+from django.db import transaction
+
+import mkt
+from mkt.access.models import Group, GroupUser
+from mkt.api.models import Access
+from mkt.users.models import UserProfile
+
+
+@transaction.commit_on_success
+def create_user(email, group_name=None, overwrite=False,
```
Comment (robhudson) [inline mkt/users/tasks.py]: Some tests around `create_user` would be good.

**Comment by robhudson** [comment]
Looks ok. Will there be a way later to associate these users with apps or anything?

File: mkt/users/tasks.py (hunk 2 above)
Comment (washort) [inline mkt/users/tasks.py]: I'm not particular about where it should go; suggestions?

**Comment by washort** [comment]
What features would you like wrt associating users with apps?

File: mkt/users/tasks.py (hunk 2 above)
Comment (robhudson) [inline mkt/users/tasks.py]: maybe mkt/users/utils.py? That exists already. Celery reads in the tasks.py file of each app looking for tasks so it seems odd to be here.

**Comment by robhudson** [comment]
> What features would you like wrt associating users with apps?

I was just thinking that a test developer should own a test app, and then wondering how that factors into the fake data.

**Comment by washort** [comment]
Associating users with fake apps should probably be done in followup to #2862.

**Comment by washort** [comment]
test added

**Comment by robhudson** [comment]
r+

**Pull request merged**